db/migrate/20210428122404_add_company_to_product.rb

```diff
@@ -0,0 +1,9 @@
+class AddCompanyToProduct < ActiveRecord::Migration[6.1]
+  def change
+    add_reference :products, :company, foreign_key: true
+    Product.all.each do |product|
+      product.update_attribute :company_id, product.customer.company_id
+    end
+    change_column_null :products, :company_id, false, 1
```

Comment:
Aqui da pra usar true?

```suggestion
    change_column_null :products, :company_id, false, true
```